### ISSUE TITLE
Leverage jsdoc-api?

### ISSUE THREAD

**Issue by pentacular** [orzechowskid/eslint-plugin-typelint#33]
**what would you like to see added?**

It looks like things could be much simplified by using npm jsdoc-api for jsdoc parsing, using line number information to reconcile with eslint's ast information.

Is there a reason to avoid leveraging it?

**Comment by orzechowskid** [comment]
well first of all, I didn't know it existed :)

but I took a look and it has a dependency on the `jsdoc` package, which itself depends on `@babel/parser`.  so I'm guessing each source file would have to be parsed twice (once by eslint or eslint-module-utils and once by jsdoc-api), which seems not great.  I tried a quick test from the command-line:

```javascript
const jsdoc = require('jsdoc-api');
const comment = jsdoc.explainSync({ source: `
` });
```
this took a human-perceivable amount of time to complete and required a bunch of temporary files on disk which seems not great either.

I am positive that there are optimizations to be made here, but we might want to do them ourselves (either in this package or in a separate package).
